### DISCUSSION TITLE
My next plans — any thoughts?

### DISCUSSION THREAD

**Discussion by Pamasich** [aclist/kbin-megamod#270] [Archived]
@aclist 
I've decided on what new mods and other changes I want to work on next. I'm making this post for transparency, so you've got an idea and can voice your opinions on them, if you have any.

- Pamasich/kbin-kes#15 
Mbin has moved the footer to the sidebar, allowing users to use infinite scrolling and still be able to access all the footer-exclusive pages. I want to replicate mbin's footer on kbin.
- Pamasich/kbin-kes#16
It should be possible to cancel writing a reply. Right now there's no way to dismiss the reply ui after opening it.
- Pamasich/kbin-kes#11
Pinned threads just look like regular threads currently. The goal here is to visually mark them so users know at a glance which threads are pinned and which aren't. The current plan here is to request the JSON of the top displayed posts and check them for the `stickied` property.
- Pamasich/kbin-kes#25
When a user doesn't exist on the instance yet, their page returns a 404 with no more information. I want to replace that with a more helpful page that explains the situation and offers help with federating the user.
- Pamasich/kbin-kes#10
From what I can tell, admins and mods just look like regular users currently. I want to point them out, similar to the existing OP label mod, so they're easier to recognize.
- Pamasich/kbin-kes#13
When using one of the two copy link options in the more menu, nothing seems to happen. The copying works, but there's zero feedback to the user. I want to add some basic visual feedback to those menu items.
- Pamasich/kbin-kes#30
It should be possible to customize the default sort order.
- Pamasich/kbin-kes#1
I want to move the items in the more menu out of it directly under the comment/post.
- Pamasich/kbin-kes#24
Like with mods and admins further above, I want content by followed users to be highlighted or given a label.
- Pamasich/kbin-kes#67
Changing how results are sorted doesn't currently redirect to the first page of the new sort option. The goal here is to change that so it does.
- Pamasich/kbin-kes#69
I want to expand the existing `User instance names` mod to cover more occurences of usernames, like the activity pages and All Content view. I haven't yet looked into how much work that would be, but I figure it should be rather easy.
- Pamasich/kbin-kes#70
Comments only get nested properly on kbin and mbin up to depth level 10. Visually, any reply past that is treated as a sibling, not child, to its parent comment. I'm thinking of maybe adding some more levels and otherwise using Old Reddit's approach to this issue. Another option is to leave things as they are but highlight the parent/child comments when hovering over them.
- Pamasich/kbin-kes#3
I prefer how block spoilers work on Old Reddit (sadly they're not implemented on newer versions), where they display as actual obscured blocks, not just hidden away in a details element until you open them.

I'm not sure yet which of these I'll tackle first, so I'm open to suggestions on what to prioritize.

**Comment by aclist** [comment]
I'm just going to touch on the ones that I thought had immediate utility (+ are easy to implement) for now:

> https://github.com/Pamasich/kbin-kes/issues/11
Pinned threads just look like regular threads currently. The goal here is to visually mark them so users know at a glance which threads are pinned and which aren't. The current plan here is to request the JSON of the top displayed posts and check them for the stickied property.

Pinned threads have `aria-label="Pinned"` and a thumbtack, if I'm not mistaken. It should be easy enough to use this as a jumping-off point for styling them or controlling them in other ways. My https://github.com/aclist/kbin-css rewrites these to say "Pinned" instead of the small thumbtack icon.

I think some contrasting border or background here could help. Speaking of which, I had been thinking it would be nice to be able to hide sticky threads altogether, since at times they take up most of the front page of a magazine and can be significantly out of date. 

> https://github.com/Pamasich/kbin-kes/issues/25
When a user doesn't exist on the instance yet, their page returns a 404 with no more information. I want to replace that with a more helpful page that explains the situation and offers help with federating the user.

Seems easy to implement and like a good idea.

> https://github.com/Pamasich/kbin-kes/issues/10
From what I can tell, admins and mods just look like regular users currently. I want to point them out, similar to the existing OP label mod, so they're easier to recognize.

I liked this as well. I suppose you would have to use `genericXMLRequest()` (KES utility function) to fetch the magazine sidebar and check the list of moderators. Could also think about expanding this to have user-editable tags to label users with.

> https://github.com/Pamasich/kbin-kes/issues/13
When using one of the two copy link options in the more menu, nothing seems to happen. The copying works, but there's zero feedback to the user. I want to add some basic visual feedback to those menu items.

This is nice as well, subtle but effective.

> https://github.com/Pamasich/kbin-kes/issues/30
It should be possible to customize the default sort order.

This has long been asked for by users, definitely a must IMO.

> https://github.com/Pamasich/kbin-kes/issues/67
Changing how results are sorted doesn't currently redirect to the first page of the new sort option. The goal here is to change that so it does.

This should definitely be fixed, agree.

> https://github.com/Pamasich/kbin-kes/issues/69
I want to expand the existing User instance names mod to cover more occurences of usernames, like the activity pages and All Content view. I haven't yet looked into how much work that would be, but I figure it should be rather easy.

I looked at the All Content page, and it seems to natively show instance names. Am I wrong?

**Question:** could you point out what the activity view is?

**Comment by Pamasich** [comment]
> Pinned threads have `aria-label="Pinned"` and a thumbtack, if I'm not mistaken. It should be easy enough to use this as a jumping-off point for styling them or controlling them in other ways.

Oh, I didn't notice that yet, that's good to know.

> Speaking of which, I had been thinking it would be nice to be able to hide sticky threads altogether, since at times they take up most of the front page of a magazine and can be significantly out of date.

Yeah, [I have been thinking about the hiding too.](../../../Pamasich/kbin-kes/issues/19) I considered this one a prerequisite since I thought they weren't already marked as sticky, but if they are, I think there's no reason to wait with that one. I'll put it into my next planned too.

Maybe these two could be combined actually and a mod made which groups the stickies together and allows to collapse them.

> I suppose you would have to use genericXMLRequest() (KES utility function) to fetch the magazine sidebar and check the list of moderators.

I don't think so, for the moderators. Admin / instance moderators (if those are listed anywhere) I'd probably have to make a request, yeah. But the sidebar and the moderator list therein is visible in every view I'd have to mark mods in, so wouldn't I only need to `document.querySelectorAll("section.user-list:has(a[href$='moderators']) li > a")`?

> Could also think about expanding this to have user-editable tags to label users with.

Yeah, [RES-style user tags is another one I've written down.](https://github.com/Pamasich/kbin-kes/issues/7) I'm not yet sure how best to do the settings for that one, and haven't felt a real need for it yet (the low population on the fediverse makes it way easier to remember people without them than on reddit), so it didn't make the cut for this list. But it is something I've been considering.

> I looked at the All Content page, and it seems to natively show instance names. Am I wrong?

Oh, it does. Weird, I remember seeing it not work there. Maybe I misremember it being a different view, or there was a bug or incompatibility.

> Question: could you point out what the activity view is?

I mean the one you get to by clicking Activity in a post's more dropdown, where it displays who up-/downvoted or boosted the user.

[Here's an example.](https://kbin.social/m/world@lemmy.world/t/944862/Colombia-seeks-to-join-Gaza-genocide-case-against-Israel-at/favourites)

**Comment by aclist** [comment]
> Maybe these two could be combined actually and a mod made which groups the stickies together and allows to collapse them.

It's not a bad idea, as removing them completely is the nuclear option, so grouping them into their own div that can be toggled would be nice.

> But the sidebar and the moderator list therein is visible in every view I'd have to mark mods in, so wouldn't I only need to document.querySelectorAll("section.user-list:has(a[href$='moderators']) li > a")?

I've grown so used to having the Hide Sidebar mod enabled that I forgot this information is always on the page.

> Yeah, https://github.com/Pamasich/kbin-kes/issues/7 I'm not yet sure how best to do the settings for that one, and haven't felt a real need for it yet (the low population on the fediverse makes it way easier to remember people without them than on reddit), so it didn't make the cut for this list. But it is something I've been considering.

The static moderator tag seems like a good enough start for now. It has a lot of utility and is immediately usable out of the box, whereas the tags thing is mostly fluff.

> I mean the one you get to by clicking Activity in a post's more dropdown, where it displays who up-/downvoted or boosted the user.

Thanks. I'd been looking high and low for this, because another mod references it, but wasn't sure what it was.

I have the 4.0.0 release ready to go at this point, just waiting for kbin.social's sporadic 404 errors to go away so I can make the post.

**Comment by Pamasich** [comment]
@aclist Is it intentional that the eslint config doesn't allow indenting switch cases?

**Reply by aclist** [reply]
I think the rule is not granular enough to encompass switch statements. It should be fixable if we add this: https://eslint.org/docs/latest/rules/indent#switchcase

`"indent": ["error", 4, {"SwitchCase": 1}]`

**Reply by aclist** [reply]
I thought this change was already pushed, but it wasn't, so I updated it on testing and applied the fix rules to all of the mods.

**Comment by Pamasich** [comment]
I've started working on [Blocky Spoilers](Pamasich/kbin-kes#3) today and will probably get it finished tomorrow. Majority of it is implemented, just need to test it on mbin and add the ability to hide/show spoilers by clicking on them.

**Comment by Pamasich** [comment]
@aclist I figured I'd start work on [Cancel Reply Button](Pamasich/kbin-kes#16) next, since it seems like a quick and easy task, but I'd have to insert a button into the reply forms for that. Does the MutationObserver used by KES get triggered and reload the plugin when a form gets added somewhere in the document? If not, I think I'll work on something else first.

**Reply by aclist** [reply]
It should, but I haven't explicitly checked. It should be easy to add some logging to KES' mutation catching event, insert an element via the console, and see if it triggers.
